tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -880,3 +880,44 @@ def _query_costs(self):
             r['ResourceId'] = r['ResourceId'].lower()
 
         return result_list
+
+
+class ParentFilter(Filter):
+    """
+    Meta filter that allows you to filter child resources by applying filters to their
+    parent resources.
+
+    You can use any filter supported by corresponding parent resource type.
+
+    :examples:
+
+    Find Azure KeyVault Keys from Key Vaults with ``owner:ProjectA`` tag.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: kv-keys-from-tagged-keyvaults
+            resource: azure.keyvault-keys
+            filters:
+              - type: parent
+                filter:
+                  "tag:owner": ProjectA
```

Comment:
Might not be good to use the special case as the example here, as it still isn't clear how to use a normal filter.  Presumably I'd put a `-type: filtername` under filter - but for 99.99% of people this will be the first time they have seen this tag syntax and they'll have no idea what is going on.